tests/nocover/test_interesting_origin.py::test_can_generate_specified_version is flaky

This test failed, but it looks like is should not.

Log:

```
 ================================== FAILURES ===================================
_____________ test_can_generate_specified_version[go_wrong_naive] _____________
[gw1] win32 -- Python 3.7.9 C:\hostedtoolcache\windows\Python\3.7.9\x64\python.exe
Traceback (most recent call last):
  File "D:\a\hypothesis\hypothesis\hypothesis-python\tests\nocover\test_interesting_origin.py", line 54, in test_can_generate_specified_version
    "function",
  File "D:\a\hypothesis\hypothesis\hypothesis-python\tests\common\utils.py", line 43, in inner
    func(*args, **kwargs)
  File "D:\a\hypothesis\hypothesis\hypothesis-python\tests\nocover\test_interesting_origin.py", line 61, in test_can_generate_specified_version
    given(st.integers(), st.integers())(settings(database=None)(function))()
  File "C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\hypothesis\_settings.py", line 218, in __call__
    f"{descr} has already been decorated with a settings object.\n"
hypothesis.errors.InvalidArgument: go_wrong_naive has already been decorated with a settings object.
    Previous:  settings(database=None, deadline=timedelta(milliseconds=200), derandomize=False, max_examples=100, phases=(Phase.explicit, Phase.reuse, Phase.generate, Phase.target, Phase.shrink), print_blob=True, report_multiple_bugs=True, stateful_step_count=50, suppress_health_check=(), verbosity=Verbosity.normal)
    This:  settings(database=None, deadline=timedelta(milliseconds=200), derandomize=False, max_examples=100, phases=(Phase.explicit, Phase.reuse, Phase.generate, Phase.target, Phase.shrink), print_blob=True, report_multiple_bugs=True, stateful_step_count=50, suppress_health_check=(), verbosity=Verbosity.normal)
---------------------------- Captured stdout call -----------------------------
Falsifying example: go_wrong_naive(
    a=0, b=0,
)

You can reproduce this example by temporarily adding @reproduce_failure('6.3.4', b'AAAAAAAAAA==') as a decorator on your test case
```

Original PR: https://github.com/HypothesisWorks/hypothesis/pull/2885
CI job: https://github.com/HypothesisWorks/hypothesis/runs/2014038419?check_suite_focus=true

--- a/hypothesis-python/tests/nocover/test_interesting_origin.py
+++ b/hypothesis-python/tests/nocover/test_interesting_origin.py
@@ -57,8 +57,14 @@ def go_wrong_coverup(a, b):
 )
 @flaky(max_runs=3, min_passes=1)
 def test_can_generate_specified_version(function):
+    @given(st.integers(), st.integers())
+    @settings(database=None)
+    def test_fn(x, y):
+        # Indirection to fix https://github.com/HypothesisWorks/hypothesis/issues/2888
+        return function(x, y)
+
     try:
-        given(st.integers(), st.integers())(settings(database=None)(function))()
+        test_fn()
     except MultipleFailures:
         pass
     else:

